pkgs/by-name/re/resonance/package.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, cargo
+, dbus
+, desktop-file-utils
+, fetchFromGitHub
+, glib
+, gst_all_1
+, gtk4
+, libadwaita
+, libxml2
+, makeWrapper
+, meson
+, ninja
+, nix-update-script
+, openssl
+, pkg-config
+, python3
+, python311Packages
+, rustPlatform
+, rustc
+, sqlite
+, stdenv
+, wrapGAppsHook4
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "resonance";
+  version = "unstable-2023-06-06";
+
+  src = fetchFromGitHub {
+    owner = "nate-xyz";
+    repo = "resonance";
+    rev = "97826093e22418c0efdb4e61cc75d981bb82c120";
+    hash = "sha256-DgNUjb8+2WTw91OGgFf97YL6lnODtkftYAP/c05RUPI=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    src = finalAttrs.src;
+    hash = "sha256-/v3OokClOk95GOzidBHRkUG7kjHQm35yPeC1n3PzcyM=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    dbus
+    desktop-file-utils
+    glib
+    gtk4
+    libxml2
+    makeWrapper
+    meson
+    ninja
+    pkg-config
+    python3
+    rustPlatform.cargoSetupHook
+    rustc
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    dbus
+    glib
+    gtk4
+    libadwaita
+    openssl
+    python3
+    sqlite
+  ] ++ (with gst_all_1; [
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+    gstreamer
+  ]);
+
+  postInstall = ''
+    wrapProgram $out/bin/resonance --prefix PYTHONPATH : ${python3.pkgs.makePythonPath (with python311Packages; [ tqdm mutagen loguru ])}
```

Comment:
There is also a `python*Packages.wrapPython` hook and a `pythonPath` attribute to do this. But I don't have a strong opinion on this.

Reply:
Okay, it seems the current derivation is already wrapping twice. So maybe we'd not introduce more wrappers.

Let's add wrapper args to the existing gapps wrapper hook instead:
```nix
preInstall = ''
  gappsWrapperArgs+=(--prefix PYTHONPATH : ${..})
''
```